internal/pkg/remote/client.go

```diff
@@ -155,45 +161,58 @@ func (c *Client) Send(
 	}
 
 	c.log.Debugf("Request method: %s, path: %s, reqID: %s", method, path, reqID)
-	c.lock.Lock()
-	defer c.lock.Unlock()
-	requester := c.nextRequester()
+	c.clientMu.Lock()
+	defer c.clientMu.Unlock()
 
-	req, err := requester.request(method, path, params, body)
-	if err != nil {
-		return nil, errors.Wrapf(err, "fail to create HTTP request using method %s to %s", method, path)
-	}
+	var err error
+	var req *http.Request
+	var resp *http.Response
 
-	// Add generals headers to the request, we are dealing exclusively with JSON.
-	// Content-Type / Accepted type can be override from the called.
-	req.Header.Set("Content-Type", "application/json")
-	req.Header.Add("Accept", "application/json")
-	// This header should be specific to fleet-server or remove it
-	req.Header.Set("kbn-xsrf", "1") // Without this Kibana will refuse to answer the request.
+	c.sortClients()
```

Comment:
not sure we should shuffle arrays with each request, it adds unnecessary CPU cycles. maybe it would be better to shuffle on error

Reply:
Sorting on each request will ensure we keep the previous behavior, preferring the no-error last used host to use next. Besides the number of fleet-server hosts are so small, that it's negligible the overhead to sort them.

Reply:
Could possibly optimize skipping sorting if already known to be sorted before.

Reply:
it needs to be sorted every time because the sorting conditions change. Every time a host is used it loses priority, thus a new sorting is required.

It isn't much different from the original code that would do extensive search on each call to `Send`. Besides the host list is so small that i's negligible the impact on performance.